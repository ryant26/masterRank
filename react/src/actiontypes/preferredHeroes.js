export const ADD_HERO = 'preferredHero/ADD_HERO';