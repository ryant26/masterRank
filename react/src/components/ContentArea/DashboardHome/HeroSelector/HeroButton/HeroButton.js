import React from 'react';
import PropTypes from 'prop-types';
import HeroImage from '../../../../Images/HeroImage/HeroImage';
const classNames = require('classnames');

const SelectorButton = ({heroName, selected, disabled, onClick}) => {

  let classes = classNames({
      HeroButton: true,
      selected: selected,
      disabled: disabled
  });

  let clickHandler = () => {
      onClick(heroName);
  };

  return (
    <div className={classes} onClick={clickHandler}>
      <HeroImage heroName={heroName}/>
    </div>
  );
};

SelectorButton.propTypes = {
  heroName: PropTypes.string.isRequired,
  selected: PropTypes.bool.isRequired,
  disabled: PropTypes.bool,
  onClick: PropTypes.func
};

SelectorButton.defaultProps = {
    disabled: false
};

export default SelectorButton;