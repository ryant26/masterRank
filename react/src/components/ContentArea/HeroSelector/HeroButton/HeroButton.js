import React from 'react';
import PropTypes from 'prop-types';
import HeroImage from '../../../HeroImage/HeroImage';
const classNames = require('classnames');

const SelectorButton = ({heroName, selected, onClick}) => {

  let classes = classNames({
      HeroButton: true,
      selected: selected,
  });

  let clickHandler = () => {
      onClick(heroName);
  };

  return (
    <div className={classes} onClick={clickHandler}>
      <HeroImage heroName={heroName}/>
    </div>
  );
};

SelectorButton.propTypes = {
  heroName: PropTypes.string.isRequired,
  selected: PropTypes.bool.isRequired,
  onClick: PropTypes.func
};

export default SelectorButton;