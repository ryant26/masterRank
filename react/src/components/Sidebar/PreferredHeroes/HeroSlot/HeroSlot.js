import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import HeroImage from "../../../Images/HeroImage/HeroImage";

const HeroSlot = ({slotNumber, selectedSlotNumber, preferredHeroes, onSlotSelected}) => {

    const classes = classnames({
        'HeroSlot': true,
        flex: true,
        'align-center': true,
        'justify-center': true,
        selected: slotNumber === selectedSlotNumber
    });

    const onClick = function() {
        onSlotSelected(slotNumber);
    };

    const content = preferredHeroes[slotNumber - 1] ? <HeroImage heroName={preferredHeroes[slotNumber - 1]}/> : slotNumber;

    return (
        <div className={classes} onClick={onClick}>{content}</div>
    );
};

HeroSlot.propTypes = {
    slotNumber: PropTypes.number.isRequired,
    selectedSlotNumber: PropTypes.number.isRequired,
    preferredHeroes: PropTypes.array.isRequired,
    onSlotSelected: PropTypes.func.isRequired
};

export default HeroSlot;