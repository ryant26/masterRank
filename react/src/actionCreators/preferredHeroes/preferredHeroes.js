import * as PreferredHeroActionsTypes from 'actiontypes/preferredHeroes';

export const addHero = (hero, preference) => {
  return {
    type: PreferredHeroActionsTypes.ADD_HERO,
    hero,
    preference
  };
};

export const removeHero = (hero, preference) => {
    return {
        type: PreferredHeroActionsTypes.REMOVE_HERO,
        hero,
        preference
    };
};

export const updateHeroes = (heroes) => {
    return {
        type: PreferredHeroActionsTypes.UPDATE_HEROES,
        heroes
    };
};