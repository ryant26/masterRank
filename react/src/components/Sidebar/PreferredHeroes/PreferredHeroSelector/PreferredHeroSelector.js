import React, { Component } from 'react';
import HeroSelector from '../../../ContentArea/DashboardHome/HeroSelector/HeroSelector';
import PropTypes from 'prop-types';
import HeroSlot from '../HeroSlot/HeroSlot';

class PreferredHeroSelector extends Component {

    constructor(props) {
        super(props);
        this.state = {
            selectedHero: props.preferredHeroes[props.selectedHeroSlot - 1],
            heroForLastAction: '',
            lastActionTaken: ''
        };

        this.maxSlots = 5;
        this.setSelectedHero = this.setSelectedHero.bind(this);
        this.setSelectedSlot = this.setSelectedSlot.bind(this);
        this.setHeroPreference = this.setHeroPreference.bind(this);
        this.clearHeroPreference = this.clearHeroPreference.bind(this);
    }

    componentWillReceiveProps(nextProps) {
        this.setSelectedHero(this.props.preferredHeroes[nextProps.selectedHeroSlot -1]);
    }

    setSelectedHero(heroName) {
        this.setState({
            selectedHero: heroName
        });
    }

    clearHeroPreference() {
        this.setState({
            heroForLastAction: this.state.selectedHero,
            lastActionTaken: `has been removed`
        });

        this.props.clearHeroPreference();
    }

    setHeroPreference(heroName) {
        if(!this.props.preferredHeroes.includes(heroName)){
            this.props.setHeroPreference(heroName);

            this.setState({
                heroForLastAction: heroName,
                lastActionTaken: `saved as preferred hero #${this.props.selectedHeroSlot}`
            });

            this.setSelectedSlot(this.props.selectedHeroSlot + 1);
        }
    }

    setSelectedSlot(slot) {
        this.props.setSelectedHeroSlot(slot);
    }

    render() {
        let heroSlots = [];
        for (let i = 1; i <= this.maxSlots; i++) {
            heroSlots.push(
                <HeroSlot
                    key={i}
                    slotNumber={i}
                    selectedSlotNumber={this.props.selectedHeroSlot}
                    preferredHeroes={this.props.preferredHeroes}
                    onSlotSelected={this.setSelectedSlot}
                />
            );
        }

        return (
            <div className="PreferredHeroSelector flex flex-column">
                <div className="header flex justify-between align-center">
                    <div>
                        <h3>Preferred Heroes</h3>
                        <div className="sub-title">Choose up to five preferred heroes to play.</div>
                    </div>
                    <div className="flex justify-between hero-slots">
                        {heroSlots}
                    </div>
                </div>
                <div className="body">
                    <HeroSelector selectedHeroes={[this.state.selectedHero]}
                                  disabledHeroes={this.props.preferredHeroes.filter((hero) => hero !== this.state.selectedHero)}
                                  onHeroSelected={this.setHeroPreference}/></div>
                <div className="footer flex align-center">
                    <div className="last-action-taken">
                        <b>{this.state.heroForLastAction.charAt(0).toUpperCase() + this.state.heroForLastAction.slice(1)}</b>{this.state.lastActionTaken}
                    </div>
                    <div className="button-group flex">
                        {this.props.preferredHeroes[this.props.selectedHeroSlot - 1] ? (
                            <div className="flex align-center button-six" onClick={this.clearHeroPreference}>
                                <div className="button-content">
                                    REMOVE
                                </div>
                            </div>
                        ) : undefined
                        }
                        {this.props.preferredHeroes.length ? (
                            <div className="save-button button-primary flex align-center" onClick={this.props.closeModal}>
                                <div className="button-content">
                                    SAVE
                                </div>
                            </div>
                        ) : undefined}

                    </div>
                </div>
            </div>
        );
    }
}

PreferredHeroSelector.propTypes = {
    selectedHeroSlot: PropTypes.number.isRequired,
    preferredHeroes: PropTypes.array.isRequired,
    setSelectedHeroSlot: PropTypes.func.isRequired,
    setHeroPreference: PropTypes.func.isRequired,
    clearHeroPreference: PropTypes.func.isRequired,
    closeModal: PropTypes.func.isRequired
};

export default PreferredHeroSelector;
