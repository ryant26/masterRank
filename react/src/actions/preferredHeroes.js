import * as PreferredHeroActionsTypes from '../actiontypes/preferredHeroes';

export const addHero = (hero, preference) => {
  return {
    type: PreferredHeroActionsTypes.ADD_HERO,
    hero,
    preference
  };
};

export const setSelectedSlot = (slot) => {
    return {
        type: PreferredHeroActionsTypes.SET_SELECTED_SLOT,
        slot
    };
};