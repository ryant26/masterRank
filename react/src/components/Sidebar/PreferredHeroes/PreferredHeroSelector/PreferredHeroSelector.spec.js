import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import {createStore} from '../../../../model/store';
import Model from '../../../../model/model';
import {addHero} from '../../../../actions/preferredHeroes';
import HeroButton from '../../../HeroSelector/HeroButton/HeroButton';
import HeroSlot from '../HeroSlot/HeroSlot';

const MockSocket = require('socket-io-mock');

import PreferredHeroSelector from './PreferredHeroSelector';

let renderComponent = (store) => {
    return renderer.create(
        <PreferredHeroSelector store={store} selectedHeroSlot={1} done={() => {}} changeSelectedHeroSlot={() => {}}/>
    );
};

describe('<PreferredHeroSelector/>', () => {
    let wrapper;
    let store;

    beforeEach(() => {
        store = createStore();
        let mockSocket = new MockSocket();
        mockSocket.addHero = () => {};
        Model.initialize(mockSocket, store);
        wrapper = mount(
            <PreferredHeroSelector store={store} done={() => {}} changeSelectedHeroSlot={() => {}}/>
        );
    });

    it('Should render without exploding', () => {
        const PreferredHeroSelectorNode = wrapper.find(PreferredHeroSelector);
        expect(PreferredHeroSelectorNode.length).toBeTruthy();
    });

    it('Should render in the correct format without any preferred heroes', () => {
        const component = renderComponent(store);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should render in the correct format with preferred heroes', () => {
        store.dispatch(addHero('genji', 1));
        store.dispatch(addHero('tracer', 2));
        store.dispatch(addHero('winston', 3));
        store.dispatch(addHero('mei', 4));
        store.dispatch(addHero('phara', 5));

        const component = renderComponent(store);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should set selectedHero in the state', () => {
        wrapper.find(HeroButton).at(0).simulate('click');
        expect(wrapper.find('PreferredHeroSelector').instance().state.selectedHero).toEqual('ana');
    });

    it('Should reflect the saved hero in the props', () => {
        wrapper.find(HeroButton).at(0).simulate('click');
        wrapper.find('button.button-primary').simulate('click');

        expect(wrapper.find('PreferredHeroSelector').instance().props.preferredHeroes).toEqual(['ana']);
    });

    it('Should reflect the selected hero slot in the props', () => {
        store.dispatch(addHero('genji', 1));
        store.dispatch(addHero('tracer', 2));
        store.dispatch(addHero('winston', 3));
        store.dispatch(addHero('mei', 4));
        store.dispatch(addHero('phara', 5));

        wrapper.find(HeroSlot).at(2).simulate('click');
        expect(wrapper.find('PreferredHeroSelector').instance().props.selectedHeroSlot).toEqual(3);
    });
});