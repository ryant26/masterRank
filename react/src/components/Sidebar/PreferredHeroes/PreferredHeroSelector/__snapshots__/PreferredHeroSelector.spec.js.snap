// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreferredHeroSelector/> Should render in the correct format with preferred heroes 1`] = `
<div
  className="PreferredHeroSelector flex flex-column"
>
  <div
    className="header flex justify-between align-center"
  >
    <div>
      <h3>
        Preferred Heroes
      </h3>
      <div
        className="sub-title"
      >
        Choose up to five preferred heroes to play.
      </div>
    </div>
    <div
      className="flex justify-between hero-slots"
    >
      <div
        className="HeroSlot flex align-center justify-center selected"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="doomfist icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroSlot flex align-center justify-center"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="genji icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroSlot flex align-center justify-center"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="mccree icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroSlot flex align-center justify-center"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="pharah icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroSlot flex align-center justify-center"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="reaper icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="body"
  >
    <div
      className="HeroSelector flex wrap"
    >
      <div
        className="HeroButton selected"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="doomfist icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton disabled"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="genji icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton disabled"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="mccree icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton disabled"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="pharah icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton disabled"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="reaper icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="soldier76 icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="sombra icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="tracer icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="bastion icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="hanzo icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="junkrat icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="mei icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="torbjorn icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="widowmaker icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="dva icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="orisa icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="reinhardt icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="roadhog icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="winston icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="zarya icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="ana icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="lucio icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="mercy icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="moira icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="symmetra icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="zenyatta icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer flex align-center"
  >
    <div
      className="last-action-taken"
    >
      <b>
        
      </b>
      
    </div>
    <div
      className="button-group flex"
    >
      <div
        className="flex align-center button-six"
        onClick={[Function]}
      >
        <div
          className="button-content"
        >
          REMOVE
        </div>
      </div>
      <div
        className="save-button button-primary flex align-center"
        onClick={[Function]}
      >
        <div
          className="button-content"
        >
          SAVE
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PreferredHeroSelector/> Should render in the correct format without any preferred heroes 1`] = `
<div
  className="PreferredHeroSelector flex flex-column"
>
  <div
    className="header flex justify-between align-center"
  >
    <div>
      <h3>
        Preferred Heroes
      </h3>
      <div
        className="sub-title"
      >
        Choose up to five preferred heroes to play.
      </div>
    </div>
    <div
      className="flex justify-between hero-slots"
    >
      <div
        className="HeroSlot flex align-center justify-center selected"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="HeroSlot flex align-center justify-center"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="HeroSlot flex align-center justify-center"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="HeroSlot flex align-center justify-center"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="HeroSlot flex align-center justify-center"
        onClick={[Function]}
      >
        5
      </div>
    </div>
  </div>
  <div
    className="body"
  >
    <div
      className="HeroSelector flex wrap"
    >
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="doomfist icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="genji icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="mccree icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="pharah icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="reaper icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="soldier76 icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="sombra icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="tracer icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="bastion icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="hanzo icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="junkrat icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="mei icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="torbjorn icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="widowmaker icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="dva icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="orisa icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="reinhardt icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="roadhog icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="winston icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="zarya icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="ana icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="lucio icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="mercy icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="moira icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="symmetra icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="HeroButton"
        onClick={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="zenyatta icon"
            className="HeroImage"
            onClick={undefined}
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer flex align-center"
  >
    <div
      className="last-action-taken"
    >
      <b>
        
      </b>
      
    </div>
    <div
      className="button-group flex"
    />
  </div>
</div>
`;
